### PR TITLE
React Capstone Full Project

### PR DESCRIPTION
**Pull request by masangana**
### Hi there! 👋

In this project, I built a mobile web application to check a list of metrics (numeric values) using of React and Redux. An API is selected that provides numeric data about a topic and then build the web app around it. The web app will have several pages:

- A home page with a list of items that could be filtered by some parameters.
- Another page for the item details

[Demo link](https://frabjous-bienenstitch-c3048c.netlify.app/)
[Loom Video](https://www.loom.com/share/f0cfce89d0714aaf905a1a28c08c47cf)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,112 @@
...
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/mavericks-db/math-magicians/issues).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
```
Comment (Davidosky007) [inline README.md:103]: * The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project. Kindly work on this requirement

File: src/components/details/Details.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  const { name } = useParams();
+  const dispatch = useDispatch();
+  const { country, loading } = useSelector((state) => ({
+    loading: state.loadingBar.default,
+    country: state.countries.selected,
+  }));
+
+  useEffect(() => {
+    dispatch(fetchCountry(name));
+    // fetchCountry(name);
```
Comment (Davidosky007) [inline src/components/details/Details.js]: * Kindly remove all commented code from your codebase as it makes your code bulky and unprofessional

File: src/components/details/Details.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+          <div className="pl-5">
+            <FontAwesomeIcon icon={faGear} />
+          </div>
+        </header>
+        <div className="Details-banner">
+          <div className="Details-banner-left">
+            <img src={image} alt="map view" className="App-map" />
+          </div>
+          <div className="Details-banner-right">
+            <h1 className="App-title">{All.country}</h1>
```
Comment (Davidosky007) [inline src/components/details/Details.js:51]: * Your details page doesn't work as expected i get a blank page whenever i click on a particular country and errors in the console too. Kindly fix this 🙏 . Below is a screenshot of the said error ⬇️ 
image

File: README.md [hunk 4]
```diff
@@ -1,70 +1,112 @@
...
 
-### `npm start`
+- [x] Languages: HTML, Sass and JavaScript
+- [x] Linters: Stylelint and ESLint
+- [x] Front-end library: React & Redux
+- [x] Code Editor: VS Code
 
-Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
+## Live Demo 🔗
```
Comment (Davidosky007) [inline README.md:17]: * Kindly work on the live demo link as the details page on the live link isn't working as expected 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @masangana 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 
* All tests passed ✔️ 
* Good video presentation 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/details/Details.js (hunk 2 above)
Comment (masangana) [reply src/components/details/Details.js]: Done

File: src/components/details/Details.js (hunk 3 above)
Comment (masangana) [reply src/components/details/Details.js:51]: ### @Davidosky007 

From my side everything is working correctly on local. as you can see in the screenshot.

So I don't understand the issue

Screenshot (150)
Screenshot (149)

File: README.md (hunk 4 above)
Comment (masangana) [reply README.md:17]: ### @Davidosky007 

Hello. 

Here I asked a check from more than 4 end-users around the word and they haven't any issue with the App.
nobody reported any issue

WhatsApp Image 2022-07-27 at 10 57 52 (2)
WhatsApp Image 2022-07-27 at 10 57 52
WhatsApp Image 2022-07-27 at 10 57 53
WhatsApp Image 2022-07-27 at 10 57 52 (1)

File: README.md [hunk 5]
```diff
@@ -1,70 +1,113 @@
...
 
-## Learn More
+### Install
 
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
+In the terminal, go to your file directory and run this command.
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+```
+$ git clone git@github.com:mavericks-db/math-magicians.git
```
Comment (elmar8287) [inline README.md]: - [ ] [OPTIONAL] Please, consider fixing the current project name in this section - it seems, `math-magicians` came from previous one 👍

File: README.md [hunk 6]
```diff
@@ -1,70 +1,113 @@
...
+- [Icon link](https://www.flaticon.com/fr/icones-gratuites/covid)
+- [Original design idea by Nelson Sakwa on Behance.](https://www.behance.net/sakwadesignstudio)
+- Code Reviewers
+- Coding Partners
 
-### `npm run build` fails to minify
+## 📝 License
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
+This project is [MIT](./MIT.md) licensed.
```
Comment (elmar8287) [inline README.md:113]: - [ ] [OPTIONAL] Please, consider fixing the broken link to the MIT.md file 👍
Screen Shot 2022-07-27 at 13 11 10

**Review by elmar8287 — APPROVED** [review]
Hi, @masangana.

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 

### Highlights

- Nice fixing the suggestions from the previous reviewer 💯 

### Optional suggestions

- I have made some optional comments under the review, please, fix them before merging 👍 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 6 above)
Comment (masangana) [reply README.md:113]: Fixed

File: README.md (hunk 5 above)
Comment (masangana) [reply README.md]: ### @elmar8287 
Thank's I didn't notice that.

Fixed

**Pull request merged**